### ISSUE TITLE
Why the model uses more vram in the larger size images?

### ISSUE THREAD

**Issue by iskenderkahramanoglu** [ultralytics/ultralytics#18200]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi!

When I take a test in a segmentation model from an image with a high pixel size (e.g. 4000x4000), the VRAM usage increases and it cannot take a test because VRAM is already low.

Isn't the input size of the image to the model the same (the model was trained with 960 imgsz), why is the larger size image consuming more VRAM. What can I do to avoid this other than resizing the image?

**Comment by pderrenger** [comment]
👋 Hello @iskenderkahramanoglu, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response, but don't worry—an Ultralytics engineer will assist you soon 👩‍💻👨‍💻.

We recommend checking out the [Docs](https://docs.ultralytics.com) for guidance, where you can find information on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many common questions may already be answered there.

Regarding your question about VRAM usage 📈, if this pertains to a 🐛 **Bug** or unexpected behavior, we’ll need a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it. This example will help us pinpoint the issue effectively.

If you're looking for a ❓ **Custom Training or Technical** explanation, please provide as much detail as possible, such as:
- Model details and configuration
- Dataset or image examples (if relevant)
- Training logs or error messages 

Additionally, ensure you are following the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for optimal performance.

## Upgrade

First, ensure you are using the latest version of `ultralytics`. Upgrade your environment with:

```bash
pip install -U ultralytics
```

This may resolve the issue if it has already been addressed in a recent update. Make sure your environment meets the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), including [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/).

## Environments

Testing the model in an up-to-date environment may also help. YOLO can be run in the following environments (with dependencies preinstalled):

- **Notebooks** with free GPU:  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>  
- **Google Cloud** Deep Learning VM ([GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/))  
- **Amazon** Deep Learning AMI ([AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))  
- **Docker Image** ([Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>  

## Join the Community

Connect with the Ultralytics community for discussions, knowledge sharing, and support:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧  
- [Discourse](https://community.ultralytics.com) for in-depth conversations  
- [Subreddit](https://reddit.com/r/Ultralytics) to share experiences  

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  

If this badge is **green**, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests verify the correct operation of YOLO across various platforms every 24 hours and on every commit.

Let us know if you have additional details, as they will help us assist you more effectively. Thank you! 😊

**Comment by Y-T-G** [comment]
> Isn't the input size of the image to the model the same (the model was trained with 960 imgsz), why is the larger size image consuming more VRAM. What can I do to avoid this other than resizing the image

You mean during prediction? Provide the code you're using.

**Comment by iskenderkahramanoglu** [comment]
> > Isn't the input size of the image to the model the same (the model was trained with 960 imgsz), why is the larger size image consuming more VRAM. What can I do to avoid this other than resizing the image
> 
> You mean during prediction? Provide the code you're using.

Yes, when try to predict a small picture (for example 1200x1200), there is no problem.
But if image size is bigger, it uses more VRAM and I take Cuda OOM.

I am using like here:

```
from ultralytics import YOLO

model = YOLO("car_segmentation.pt")
results = model("cars_on_the_street.png")
```

**Comment by Y-T-G** [comment]
What image size is shown in the logs?

**Comment by iskenderkahramanoglu** [comment]
> What image size is shown in the logs?

Same image size for both of them:

Image resized to half (2257x1166):
image 1/1 /content/home/0a7a65be-b427-4de7-8ed3-9cb035db9cd0067d5e03a5_resampled.jpg: 896x1696 1 car, 116.6ms
Speed: 148.7ms preprocess, 116.6ms inference, 460.5ms postprocess per image at shape (1, 3, 896, 1696)

Image full size (4514x2332):
image 1/1 /content/home/0a7a65be-b427-4de7-8ed3-9cb035db9cd0067d5e03a5.jpg: 896x1696 1 car, 129.7ms
Speed: 13.0ms preprocess, 129.7ms inference, 12.0ms postprocess per image at shape (1, 3, 896, 1696)

**Comment by Y-T-G** [comment]
Can you try predicting with `imgsz=960`?

**Comment by iskenderkahramanoglu** [comment]
> Can you try predicting with `imgsz=960`?

I misremembered the model image size, the model size is 1696. But that's not the problem, in the logs I get the same imgsz in both, but in the resampled one I can get tests, in the other one I need more VRAM.

**Comment by Y-T-G** [comment]
What do you mean by tests?

**Comment by iskenderkahramanoglu** [comment]
> What do you mean by tests?

Sorry for my bad English, I mean, I can predict with small size images (2257x1166), but I can not predict with bigger images (4514x2332) from 1696 imgsz Yolov8 model.

**Comment by Y-T-G** [comment]
@Laughing-q I think some of the tensors like the one for visualizing features should be on CPU by default since otherwise they're wasting VRAM. Also probably an option to return results on CPU too because for images with lots of objects, the masks can consume a lot of VRAM.

**Comment by Y-T-G** [comment]
Post-processing time might be higher. You can check.

**Comment by iskenderkahramanoglu** [comment]
> Post-processing time might be higher. You can check.

Okay, I will try than write the results.
Thank you so much!

**Comment by glenn-jocher** [comment]
You're welcome! Let us know how it goes or if you encounter any issues.

**Comment by Laughing-q** [comment]
> @Laughing-q I think some of the tensors like the one for visualizing features should be on CPU by default since otherwise they're wasting VRAM. Also probably an option to return results on CPU too because for images with lots of objects, the masks can consume a lot of VRAM.

ah but then users are gonna ask why the whole process is much slower if we do it. And for the OOM issue I consider it as an edge case no? basically it only happens when users do inference on large resolution images. If we could provide them some customized code to solve their issue would be perfect! Just like you already did. Nice work
